Sources/BraveWallet/Crypto/Stores/AssetDetailStore.swift

```diff
@@ -93,63 +132,118 @@ class AssetDetailStore: ObservableObject {
     walletService.defaultBaseCurrency { [self] currencyCode in
       self.currencyCode = currencyCode
     }
+    
+    fetchSupportedBuyTokens()
   }
 
   private let percentFormatter = NumberFormatter().then {
     $0.numberStyle = .percent
     $0.maximumFractionDigits = 2
   }
   
+  private func fetchSupportedBuyTokens() {
+    Task { @MainActor in
+      let buyOptions: [BraveWallet.OnRampProvider]
+      if Locale.preferredLanguages.first?.caseInsensitiveCompare("en-us") == .orderedSame {
+        buyOptions = [.ramp, .sardine, .transak]
+      } else {
+        buyOptions = [.ramp, .transak]
+      }
+      let selectedCoin = await walletService.selectedCoin()
+      selectedNetwork = await rpcService.network(selectedCoin)
+      allBuyTokensAllOptions = await blockchainRegistry.allBuyTokens(in: selectedNetwork, for: buyOptions)
```

Comment:
We need to use the coin from the asset here, otherwise we fail to display `buy` button on ETH/SOL depending on the selected network:

![buy ETH](https://user-images.githubusercontent.com/5314553/233496898-49636190-e621-4b0f-9b27-ff2561e647ff.png) | ![buy SOL](https://user-images.githubusercontent.com/5314553/233496901-37189871-db87-4ba3-a1b6-e4595c615045.png)
--|--

Reply:
there will be no coin information from token since its from the market tab. we have to use selected coin value on mobile.

Reply:
Oh yeah you are right, seeing the same on desktop.